File: backend/app/schemas/servicios.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class CrearCategoriaRequest(BaseModel):
    nombre: str
    descripcion: str | None = None
    icono_url: str | None = None
    color_hex: str = "#000000"
    orden_visualizacion: int = 0


class CategoriaResponse(BaseModel):
    id: UUID
    nombre: str
    descripcion: str | None
    icono_url: str | None
    color_hex: str
    orden_visualizacion: int
    esta_activo: bool


class ActualizarCategoriaRequest(BaseModel):
    nombre: str | None = None
    descripcion: str | None = None
    orden_visualizacion: int | None = None
    esta_activo: bool | None = None


class CrearServicioRequest(BaseModel):
    categoria_id: UUID
    nombre: str
    descripcion_tecnica: str | None = None
    duracion_minutos: int = Field(gt=0)
    precio: float = Field(gt=0)
    monto_deposito: float = Field(ge=0)
    requiere_ficha_salud: bool = False
    horas_cancelacion_sin_penalidad: int = 5
    imagen_referencia_url: str | None = None


class ActualizarServicioRequest(BaseModel):
    nombre: str | None = None
    descripcion_tecnica: str | None = None
    duracion_minutos: int | None = Field(default=None, gt=0)
    precio: float | None = Field(default=None, gt=0)
    monto_deposito: float | None = Field(default=None, ge=0)
    requiere_ficha_salud: bool | None = None
    horas_cancelacion_sin_penalidad: int | None = None
    imagen_referencia_url: str | None = None
    esta_activo: bool | None = None


class ServicioResponse(BaseModel):
    id: UUID
    categoria_id: UUID
    nombre: str
    descripcion_tecnica: str | None
    duracion_minutos: int
    precio: float
    monto_deposito: float
    requiere_ficha_salud: bool
    horas_cancelacion_sin_penalidad: int
    imagen_referencia_url: str | None
    esta_activo: bool


class HorarioDisponible(BaseModel):
    inicio: datetime
    fin: datetime


class DisponibilidadResponse(BaseModel):
    personal_id: UUID
    horarios: list[HorarioDisponible]

File: frontend/src/types/servicios.ts
export interface ICategoria {
  id: string;
  nombre: string;
  descripcion: string | null;
  icono_url: string | null;
  color_hex: string;
  orden_visualizacion: number;
  esta_activo: boolean;
}

export interface ICategoriaCreate {
  nombre: string;
  descripcion?: string;
  icono_url?: string;
  color_hex?: string;
  orden_visualizacion?: number;
}

export interface ICategoriaUpdate {
  nombre?: string;
  descripcion?: string;
  orden_visualizacion?: number;
  esta_activo?: boolean;
}

export interface IServicio {
  id: string;
  categoria_id: string;
  nombre: string;
  descripcion_tecnica: string | null;
  duracion_minutos: number;
  precio: number;
  monto_deposito: number;
  requiere_ficha_salud: boolean;
  horas_cancelacion_sin_penalidad: number;
  imagen_referencia_url: string | null;
  esta_activo: boolean;
}

export interface IServicioCreate {
  categoria_id: string;
  nombre: string;
  descripcion_tecnica?: string;
  duracion_minutos: number;
  precio: number;
  monto_deposito: number;
  requiere_ficha_salud?: boolean;
  horas_cancelacion_sin_penalidad?: number;
  imagen_referencia_url?: string;
}

export interface IServicioUpdate {
  nombre?: string;
  descripcion_tecnica?: string;
  duracion_minutos?: number;
  precio?: number;
  monto_deposito?: number;
  requiere_ficha_salud?: boolean;
  horas_cancelacion_sin_penalidad?: number;
  imagen_referencia_url?: string;
  esta_activo?: boolean;
}

export interface IHorarioDisponible {
  inicio: string;
  fin: string;
}

export interface ISlotDisponible {
  personal_id: string;
  horarios: IHorarioDisponible[];
}
